ssl/ssl_lib.c

```diff
@@ -4257,6 +4277,39 @@ void ssl_set_masks(SSL_CONNECTION *s)
 
     mask_a |= SSL_aNULL;
 
+    /*
+     * You can do anything with an RPK key, since there's no cert to restrict it
+     * But we need to check for private keys
+     */
+    if (tls12_rpk_and_privkey(s, SSL_PKEY_RSA)) {
+        mask_a |= SSL_aRSA;
+        mask_k |= SSL_kRSA;
+    }
+    if (tls12_rpk_and_privkey(s, SSL_PKEY_RSA_PSS_SIGN)) {
+        if (TLS1_get_version(&s->ssl) == TLS1_2_VERSION)
```

Comment:
Just `s->version == TLS1_2_VERSION` - seems better to not go through the SSL object? What about DTLS?

Reply:
This was based on https://github.com/openssl/openssl/blob/c4cb151b4d19f5f2eb302baafac65e36e4408381/ssl/ssl_lib.c#L4251 which also only checks for TLS1_2 (and also uses `TLS1_get_version()`).

Reply:
Just saying I'm consistent with the rest of the function here. :) It might be strange to use two different methods to look at the version, and nothing else here looks at TLSv1.2 + DTLSv1.2, it's all TLSv1.2...

Reply:
Ok. I personally think this is wrong, but I can see it as out-of-scope for this PR.

Reply:
I can write up an issue to track this?